ibm_catalog.json

```diff
@@ -0,0 +1,66 @@
+{
+  "products": [
+    {
+      "name": "terraform-ibm-secrets-manager-iam-engine",
+      "label": "IAM secrets engine module",
+      "product_kind": "module",
+      "tags": [
+        "dev_ops",
+        "target_terraform",
+        "terraform",
+        "module"
+      ],
+      "keywords": [
+        "terraform",
+        "iam",
+        "secrets",
+        "encryption"
+      ],
+      "short_description": "This module configures an IAM secrets engine for an existing secrets manager instance.",
+      "long_description": "Use this module to configure an IAM secrets engine for an existing secrets manager instance.\n#### Usage\n```hcl\nmodule \"iam_secrets_engine\" {\n  source                               = \"terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/ibm\"\n  version                              = \"X.X.X\"  # Replace \"X.X.X\" with a release version to lock into a specific release\n  resource_group_id                    = \"xxXXxxXXxXxXXXXxxXxxxXXXXxXXXXX\"\n  region                               = \"us-south\"\n  iam_engine_name                      = \"iam-engine\"\n  secrets_manager_guid                 = \"XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX\"\n}\n```",
```

Comment:
Update the description to say this:
Use this module to configure a secrets manager IAM credentials engine for an existing secrets manager instance.
^ This should probably also be updated in the description in `.github/settings.yml` to match.

Also the source needs to be updated to the catalog source url (and remove version arg). Source url would be `https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source?archive=tgz&kind=terraform&name=terraform-ibm-secrets-manager-iam-engine&version=X.X.X`

Reply:
If I grab the [raw text](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/3a6264fe03fc44e856fd3a4b966e6fcc409ed07e/ibm_catalog.json) (must be the raw text) and paste it into https://www.freeformatter.com/json-escape.html#before-output and click unescape json I see alot of unnecessary spaces before each of the equals. Is that really the case?
![image](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/assets/29608224/d3505dfe-3b12-4256-9af0-32286490a635)

(PS No need to add a final `\n` at the end, it just adds more whitespace)

Reply:
I think the long description needs more info on what the modules does. So add the bullet points that are in the module readme

Reply:
I agree with Conall that there's an opportunity here to expand on the short description in the long desc. For example, can we explain the goals of the repo (as in why it exists or maybe what this does that the [straight Terraform](https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-configure-iam-engine&interface=terraform) doesn't)? You can include markdown within the JSON for formatting. For example, you can add an unordered list. The features should be more "marketing" highlights.